lib/node_modules/@stdlib/math/base/special/gamma-lanczos-sum-expg-scaled/src/main.c

```diff
@@ -0,0 +1,74 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/gamma_lanczos_sum_expg_scaled.h"
+#include "stdlib/constants/float64/pinf.h"
+
+/**
+* Calculates the Lanczos sum for the approximation of the gamma function (scaled by `exp(-g)`, where `g = 10.900511`.
+*
+* @param {number} x - input value
+* @returns {number} Lanczos sum approximation
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( 4.0 );
+* // returns ~0.018
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( -1.5 );
+* // returns ~25.337
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( -0.5 );
+* // returns ~-12.911
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( 0.5 );
+* // returns ~1.772
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( 0.0 );
+* // returns Infinity
+*
+* @example
+* var v = gammaLanczosSumExpGScaled( NaN );
+* // returns NaN
+*/
+double stdlib_base_gamma_lanczos_sum_expg_scaled( const double x ) {
+	double ax;
+	double s1;
+	double s2;
+	double x1;
+	if ( x == 0.0 ) {
+		return STDLIB_CONSTANT_FLOAT64_PINF;
+	}
+	if ( x < 0.0 ) {
+		ax = -x;
+	} else {
+		ax = x;
+	}
+	if ( ax <= 1.0 ) {
```

Comment:
Same comment as in https://github.com/stdlib-js/stdlib/pull/1834#discussion_r1520513679

Reply:
If the issue is L58, we should update `evalrational-compile-c` to handle infinities.

Reply:
Thanks for these suggestions. Once we are done with the PR for [`gamma-lanczos-sum`](https://github.com/stdlib-js/stdlib/pull/1834), then I will make changes according to that here, so that things aren't repetitive.